console/packages/components/src/components/tabs/Tabbar.vue

```diff
@@ -52,23 +77,151 @@ function handleHorizontalWheel(event: WheelEvent) {
   }
 }
 
+// 保存每项 tab 宽度
+function calculateItemWidth(
+  n: Record<string, string>[] | undefined,
+  o: Record<string, string>[] | undefined
+) {
+  if (!tabbarItemsRef.value) return;
+  if (
+    tabItemRefs.value &&
+    tabItemRefs.value.length === n?.length &&
+    n?.length !== o?.length
+  ) {
+    for (const item of tabItemRefs.value) {
+      itemWidthArr.value.push(item.offsetWidth);
+    }
+    arrowFlag.value = !arrowFlag.value;
+  }
+}
+
```

Comment:
```suggestion
const itemWidthArr = computed(() => {
  const itemWidthArr: number[] = [];
  if (!tabItemRefs.value) return itemWidthArr;
  for (const item of tabItemRefs.value) {
    itemWidthArr.push(item.offsetWidth);
  }
  return itemWidthArr;
});
```

items 在初始化之后数量不会发生变化，因此不需要监听 items 的数量。可以按照上述使用 `computed` 获取或者其他方式获取一次即可。

Reply:
这里不太对，异步请求回来的 items 会有变化吧，试试 Hao 这个主题的 Tabbar

Reply:
> 这里不太对，异步请求回来的 items 会有变化吧，试试 Hao 这个主题的 Tabbar

items 只有在切换主题或者重载主题配置时计算。

https://github.com/halo-dev/halo/blob/8b405faa5790304f39e9a1d3455f25ce6c9b6ddf/console/console-src/modules/interface/themes/layouts/ThemeLayout.vue#L96-L110

不过对于组件而言，监听 items 倒是没有问题的。不过在此场景下，我建议使用 `computed` 来计算 tabItemRefs 性能更好一些。或者只单独 watch tabItemRefs 而无需关心 items 。

Reply:
computed 不行，初始箭头状态拿不到。改为单独 watch tabItemRefs